packages/commerce-sdk-react/src/test-utils.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {render, RenderOptions} from '@testing-library/react'
+import nock from 'nock'
+import React from 'react'
+import CommerceApiProvider from './provider'
+
+const sampleProps = {
+    proxy: 'http://localhost:3000/mobify/proxy/api',
+    clientId: 'c9c45bfd-0ed3-4aa2-9971-40f88962b836',
+    organizationId: 'f_ecom_zzrf_001',
+    shortCode: '8o7m175y',
+    siteId: 'RefArchGlobal',
+    locale: 'en_US',
+    currency: 'USD'
+}
+const TestProviders = (props: {children: React.ReactNode}) => {
+    return (
+        <CommerceApiProvider
+            {...sampleProps}
+            queryClientConfig={{
+                defaultOptions: {queries: {retry: false}, mutations: {retry: false}}
+            }}
```

Comment:
@alexvuong this is how I disable the `retry` (during testing only). I did it globally by passing in a config object for the Query Client, which requires our `CommerceApiProvider` to accept an additional prop.

Reply:
Turning off `retry` is actually suggested in React Query's doc too: https://react-query-v3.tanstack.com/guides/testing#turn-off-retries

> The library defaults to three retries with exponential backoff, which means that your tests are likely to timeout if you want to test an erroneous query. The easiest way to turn retries off is via the QueryClientProvider.